### ISSUE TITLE
parser: for-of loop with labelled function statement in non-strict mode should throw syntax error

### ISSUE THREAD

**Issue by Dunqing** [oxc-project/oxc#10051]
[Test262 test]( https://github.com/tc39/test262/blob/bc5c14176e2b11a78859571eb693f028c8822458/test/language/statements/for-of/labelled-fn-stmt-lhs.js) | [Babel Playground](https://babeljs.io/repl#?browsers=defaults%2C%20not%20ie%2011%2C%20not%20ie_mob%2011&build=&builtIns=false&corejs=3.21&spec=false&loose=false&code_lz=PTAEGEHsAcE8CcCWBzAFgF1ACnASlAEwAMAjAGyjqoCmoAagByjTyQBW1AxpgIYCuVSPADOAOlABBADZTQSNOmFzqw6vABu1ACaiAUCFAAVVIiWdIW2qdDJIm-ADttoAEaxKNUACEAygBFQKUROagdVUAAzSD4HLVBEBw9aABkASXAAUQA5HwzIxClqPWAAKgBaCt1LYU4kaHRESAcALlBUzGseUB9YB3QeAA9QDPhWeHiItuFknhdqGW0AMRjuRocsH370agBbUPR8a3R4PiLdJ2QeBs1m3VBmVB5VVugeEWo7ylhoalaevsGIzGuhUiC0rVUnDKqh2PD6wWE0K2wWhuzhDU4iOobyksDKajGwhBwjIYNaJAAzKIAOyiEh03QJKKtAA-n3u0kEfDQkgcTiGXlA1VqiDmSjhoGoA22YTWoAA7iZOKhmGodohFIFZvNCnEIisGk12UKuFI3lc1kpooowbRIJNhMdgpgdhZqAAaDzWbHwXGS0ZCeLi6DQILOeDUS7wLSFYRE-73e1JLUOZB8HjIWiuyx6CplErAXS6AAkfgA8lky4YMnQJMkAKoSatYXAAbiLUXGWCGSYA2gBdfBmuZSEitYfzAitfUOVZNSIt0AAbwAvrogA&debug=false&forceAllTransforms=false&modules=false&shippedProposals=false&evaluate=false&fileSize=false&timeTravel=false&sourceType=script&lineWrap=true&presets=&prettier=true&targets=&version=7.27.0&externalPlugins=%40babel%2Fplugin-transform-object-rest-spread%407.25.9&assumptions=%7B%7D) | [OXC Playground](https://playground.oxc.rs/#eNpNU8Fu2zAM/RVC2CEpknTpsG7wTmmbAgWKdliyXuYdFJu2VciUIcptgi7/PkpO0p4kSuJ7j4/UmypUps7P4dp1O2/qJsDoegwXn+eXEBqEp+/QefeMRQDdh8Z5nsHCWkhPGTwy+hcsZzkJxroxDIUrEWSt3Qt6whI2u4R0tboBawokRqhcTyUYShf3d9fLh9USKmMx4pxNp9OcSuTCmy4YRxnchYioYbWjoLew9N55MBXc8b3eoLVY3vZUxLejVdABW6QwjinB9xGTsNbBvGCWE0DXaMYMOu0ZYxx2nYQDdELOCdmUGTAWU8ZWUzAFT1mAzccD1N7uphgzOKZcxpz5l9m32Xw2z8lQ5TL4FxlALBPv+rqBhWjZwhUM5W1QqiLAbRBbRD28NqZooEPfGnHXHoqD6lDdAFZiYbUXNY4YXB9ELIKrgIM30qdWGjARZ6X8JBGSxGRg11kjcF7s8KVFFt0RUHJjI6ymutc1JgRxTfpwdp4L6aebx4fH9fJpcf97sV6Oxj/iYSWYo23M/fN3PEidZ8N6kZ0UQzUaw9s+JzVRTmVvyvcUF052qyw2aKKsoXDcc+E6PAW7duPsMQpeEwtvq7JKW8b9RKUu+oiorXWvvzD0nh4HT44zcUzvDtP6U3ux+0Thel/grQwf6VaIVUAOs2feiuLhbi0DIufDPCohjXITqexPmgYVQfsapRaFLH/oq2AYdlZGsrz50LV3/fG31JgsEf6NxVWibHU36JMnrSFTmSNf4Sh4Z2+l2Jgkf2zjWOQNgPv9f5ODVuI=)

```js
// Copyright (C) 2016 the V8 project authors. All rights reserved.
// This code is governed by the BSD license found in the LICENSE file.
/*---
description: It is a Syntax Error if IsLabelledFunction(Statement) is true.
negative:
  phase: parse
  type: SyntaxError
esid: sec-semantics-static-semantics-early-errors
es6id: 13.7.1.1
info: |
    Although Annex B describes an extension which permits labelled function
    declarations outside of strict mode, this early error is applied regardless
    of the language mode.
---*/

$DONOTEVALUATE();

for (x of []) label1: label2: function f() {}

```

### Expect

Both should throw a syntax error in non-strict mode and strict mode

### Actual

Non-strict mode has no error, and strict mode has a syntax error

### Additional

Ditto for `IfStatement`, `WhileStatement`, `DoWhileStatement`, `ForStatement`,`ForInStatement` and `WithStatement`
